### ISSUE TITLE
Incorrect MO response (dC/dR) from implicit SCF differentiation for CH₂O

### ISSUE THREAD

**Issue by wsr1999** [fishjojo/pyscfad#98]
# Incorrect MO response (dC/dR) from implicit SCF differentiation for CH₂O

The nuclear gradient of `mf.e_tot` is correct, but derivatives of `mf.mo_coeff` propagated to downstream quantities are wrong for CH₂O/STO-3G. This affects both user-written post-SCF code and PySCFAD's own `tdscf.rhf.CIS`.

Works correctly for H₂ and H₂O.

## Environment

pyscfad 0.2.0, pyscf 2.12.1, jax 0.8.3

## 1. CIS gradient

```python
import jax
from pyscfad import gto, scf, tdscf

mol = gto.Mole()
mol.atom = 'C 0 0 -0.5265; O 0 0 0.6644; H 0 0.9369 -1.1129; H 0 -0.9369 -1.1129'
mol.unit = 'Angstrom'; mol.basis = 'sto-3g'
mol.build(trace_exp=False, trace_ctr_coeff=False)

def energy(mol):
    mf = scf.RHF(mol)
    e_hf = mf.kernel()
    mytd = tdscf.rhf.CIS(mf)
    mytd.nstates = 3
    e = mytd.kernel()[0]
    return e[0] + e_hf

grad = jax.grad(energy)(mol)
print(grad.coords)
```

AD vs FD vs PySCF analytical (`td.nuc_grad_method()`):

```
         AD (PySCFAD)   FD (PySCF)   Analytical (PySCF)
C z:      0.30519978    0.26445078       0.26445076
O z:     -0.29238390   -0.23956405      -0.23956404
H y:      0.00946028    0.01405617       0.01405616
H z:     -0.00640794   -0.01244336      -0.01244336
```

## 2. tr(P²) where P = mf.make_rdm1()

```python
import jax
from jax import numpy as jnp
from pyscfad import gto, scf

mol = gto.Mole()
mol.atom = 'C 0 0 -0.5265; O 0 0 0.6644; H 0 0.9369 -1.1129; H 0 -0.9369 -1.1129'
mol.unit = 'Angstrom'; mol.basis = 'sto-3g'
mol.build(trace_exp=False, trace_ctr_coeff=False)

def density_trace(mol):
    mf = scf.RHF(mol)
    mf.verbose = 0; mf.conv_tol = 1e-12; mf.kernel()
    P = mf.make_rdm1()
    return jnp.trace(P @ P)

grad = jax.grad(density_trace)(mol)
print(grad.coords)
```

```
         AD (PySCFAD)   FD (PySCF)       Error
C z:     -0.03172991   -0.09817850     6.64e-02
O z:      0.58075109    0.66696767     8.62e-02
H y:      0.32811289    0.33564065     7.53e-03
H z:     -0.27451059   -0.28439458     9.88e-03
```

(Only non-zero components shown. x-components are zero by symmetry.)

## Notes

- `jax.grad` of `mf.e_tot` alone is correct to ~1e-9 for the same molecule.
- H₂ and H₂O give errors below 1e-8 for both tests above.
- All FD values verified independently with pure PySCF (no PySCFAD involvement).
- The description of this issue was refined by generative AI.

**Comment by fishjojo** [comment]
Please use the implicit differentiation solver for mean-field by
```
from pyscfad import config
config.update("pyscfad_scf_implicit_diff", True)
```
and the following input
```
import jax
from jax import numpy as jnp
from pyscfad import gto, scf, tdscf, config

config.update("pyscfad_scf_implicit_diff", True)

mol = gto.Mole()
mol.atom = 'C 0 0 -0.5265; O 0 0 0.6644; H 0 0.9369 -1.1129; H 0 -0.9369 -1.1129'
mol.basis = 'sto-3g'
mol.build(trace_exp=False, trace_ctr_coeff=False)

def energy(mol):
    mf = scf.RHF(mol)
    e_hf = mf.kernel()
    mytd = tdscf.rhf.CIS(mf)
    mytd.nstates = 3
    e = mytd.kernel()[0]
    return e[0] + e_hf

jac = jax.grad(energy)(mol)
print(jac.coords)

def density_trace(mol):
    mf = scf.RHF(mol)
    mf.verbose = 0; mf.conv_tol = 1e-12; mf.kernel()
    P = mf.make_rdm1()
    return jnp.trace(P @ P)

grad = jax.grad(density_trace)(mol)
print(grad.coords)
```
gives the correct results
```
converged SCF energy = -112.352758703772
Excitation energies (eV)
[ 4.44316979  9.75497513 12.72154933]
[[-4.25531524e-17 -5.04998696e-19  2.64450677e-01]
 [ 6.89177816e-17  1.78992074e-16 -2.39563932e-01]
 [-7.37556416e-17  1.40561620e-02 -1.24433723e-02]
 [ 4.73910125e-17 -1.40561620e-02 -1.24433723e-02]]
[[-6.07839031e-16  4.94761493e-16 -9.81784720e-02]
 [ 6.48858802e-16 -2.72071647e-15  6.66967726e-01]
 [-2.38205278e-16  3.35640638e-01 -2.84394627e-01]
 [ 1.97185507e-16 -3.35640638e-01 -2.84394627e-01]]
```

**Comment by wsr1999** [comment]
Thanks for the quick response! Setting pyscfad_scf_implicit_diff = True fixes the issue.
